lib/internal/util.js

```diff
@@ -560,6 +562,14 @@ function SideEffectFreeRegExpPrototypeExec(regex, string) {
   return FunctionPrototypeCall(RegExpFromAnotherRealm.prototype.exec, regex, string);
 }
 
+function isArrayBufferDetached(value) {
+  if (ArrayBufferPrototypeGetByteLength(value) === 0) {
+    return _isArrayBufferDetached(value);
```

Comment:
Crossing the JS-C++ boundary is gonna be slower than the current implementation although I'm not sure if this call in the zero length case is gonna be the primary bottleneck for any user code.

Did you consider implementing @jasnell's suggestion in https://github.com/nodejs/node/pull/45512#discussion_r1027103978? Seems like that approach would solve this problem without adding any performance overhead in any case.

> One thing we might be able to do is add a hidden symbol to ArrayBuffer instances when we detach them in c++ land. Then we can have an isDetached check in js that checks the length and the presence of that symbol.

Reply:
IMHO, that is a really good solution for performance but I believe it's too hacky to maintain in the long term. I can't think of a good/maintainable way to enforce attaching the hidden symbol as a rule.